File: backend/app/config/concept_types.py
"""
Централизиран модул за всички типове концепции и съобщения в pi_sb.

ЕДИНСТВЕН източник на истина: config/concept_types.json
Ако добавяш нов тип, промени само JSON файла — този модул го чете автоматично.

Употреба:
    from app.config.concept_types import CONCEPT_TYPES, CONCEPT_TYPE_DESCRIPTIONS, NOTIFICATION_TYPES
"""

import json
from pathlib import Path
from typing import Optional

from app.config.settings import get_app_internal_dir

# ═══════════════════════════════════════════════════════════════
# Зареждане от JSON конфиг
# ═══════════════════════════════════════════════════════════════

_CONFIG_PATH = get_app_internal_dir() / "config" / "concept_types.json"


def _load_config() -> dict:
    """Зарежда конфигурацията от JSON файла."""
    if not _CONFIG_PATH.exists():
        raise FileNotFoundError(
            f"Конфигурационният файл не е намерен: {_CONFIG_PATH}\n"
            f"Създай config/concept_types.json с валидни типове."
        )
    with open(_CONFIG_PATH, "r", encoding="utf-8") as f:
        return json.load(f)


_config = _load_config()

# ═══════════════════════════════════════════════════════════════
# OKF концепции (type поле)
# ═══════════════════════════════════════════════════════════════

CONCEPT_TYPES: list[str] = list(_config["types"].keys())

# Default тип за нови концепции (използва се при fallback/невалиден LLM отговор)
DEFAULT_CONCEPT_TYPE: str = _config.get("default_type", CONCEPT_TYPES[2])

# ═══════════════════════════════════════════════════════════════
# Български описания за всеки тип (за индексите и UI)
# ═══════════════════════════════════════════════════════════════

CONCEPT_TYPE_DESCRIPTIONS: dict[str, str] = dict(_config["types"])

# ═══════════════════════════════════════════════════════════════
# Фронтенд нотификации (типове съобщения за UI)
# ═══════════════════════════════════════════════════════════════

NOTIFICATION_TYPES: list[str] = list(_config.get("notifications", []))

# ═══════════════════════════════════════════════════════════════
# API response статуси
# ═══════════════════════════════════════════════════════════════

API_STATUS_TYPES: list[str] = list(_config.get("api_statuses", []))

File: frontend/src/config/conceptTypes.ts
/**
 * Централизиран модул за типове концепции и съобщения.
 * 
 * ЕДИНСТВЕН източник на истина: config/concept_types.json
 * Ако добавяш нов тип, промени само JSON файла — този модул го чете автоматично.
 */

// ══════════════════════════════════════════════════
// Зареждане от JSON конфиг
// ══════════════════════════════════════════════════

// Vite поддържа директен import на JSON
import config from '../../../config/concept_types.json';

interface ConceptTypesConfig {
    types: Record<string, string>;
    notifications: string[];
    api_statuses: string[];
    default_type: string;
}

const typedConfig = config as ConceptTypesConfig;

// ══════════════════════════════════════════════════
// OKF концепции (type поле)
// ══════════════════════════════════════════════════

export const CONCEPT_TYPES: string[] = Object.keys(typedConfig.types);

/** Default тип за нови концепции */
export const DEFAULT_CONCEPT_TYPE: string = typedConfig.default_type || CONCEPT_TYPES[2];

// ══════════════════════════════════════════════════
// Български описания за всеки тип
// ══════════════════════════════════════════════════

export const CONCEPT_TYPE_DESCRIPTIONS: Record<string, string> = { ...typedConfig.types };

// ══════════════════════════════════════════════════
// Типове за UI нотификации
// ══════════════════════════════════════════════════

export type NotificationType = 'success' | 'error' | 'warning' | 'info';

export const NOTIFICATION_TYPES: NotificationType[] = [...typedConfig.notifications] as NotificationType[];

// ══════════════════════════════════════════════════
// API response статуси
// ══════════════════════════════════════════════════

export type ApiStatusType = 'ok' | 'error' | 'warning';

export const API_STATUS_TYPES: ApiStatusType[] = [...typedConfig.api_statuses] as ApiStatusType[];

// ══════════════════════════════════════════════════
// Helper functions
// ══════════════════════════════════════════════════

/** Връща българското описание за даден тип концепция */
export function getConceptTypeDescription(type: string): string {
    return CONCEPT_TYPE_DESCRIPTIONS[type] || `${type.replace(/_/g, ' ').replace(/\b\w/g, (c) => c.toUpperCase())} — автоматично`;
}

/** Проверява дали даден тип е валиден */
export function isValidConceptType(type: string): boolean {
    return CONCEPT_TYPES.includes(type);
}
